crates/shared/src/price_estimation/native/coingecko.rs

```diff
@@ -25,6 +27,18 @@ pub struct CoinGecko {
     base_url: Url,
     api_key: Option<String>,
     chain: String,
+    native_price: NativePrice,
+}
+
+enum NativePrice {
+    Eth,
+    Other(Token),
+}
+
+lazy_static! {
```

Comment:
lazily initialized, so we only initialize it for the xdai network

Reply:
This is already part of the config somewhere (e.g. [here](https://github.com/cowprotocol/services/blob/cd2af46fa97d17c0d547e4082075837cea916ee0/crates/autopilot/src/infra/blockchain/contracts.rs#L6)). no need to hardcode this